src/libs/kata-types/src/mount.rs

```diff
@@ -0,0 +1,86 @@
+// Copyright (c) 2019-2021 Alibaba Cloud
+// Copyright (c) 2019-2021 Ant Group
+//
+// SPDX-License-Identifier: Apache-2.0
+//
+
+/// Prefix to mark a volume as Kata special.
+pub const KATA_VOLUME_TYPE_PREFIX: &str = "kata:";
+
+/// The Mount should be ignored by the host and handled by the guest.
+pub const KATA_GUEST_MOUNT_PREFIX: &str = "kata:guest-mount:";
+
+/// KATA_EPHEMERAL_DEV_TYPE creates a tmpfs backed volume for sharing files between containers.
+pub const KATA_EPHEMERAL_VOLUME_TYPE: &str = "kata:ephemeral";
+
+/// KATA_HOST_DIR_TYPE use for host empty dir
+pub const KATA_HOST_DIR_VOLUME_TYPE: &str = "kata:hostdir";
+
+/// Information about a mount.
+#[derive(Debug, Clone, Default, serde::Serialize, serde::Deserialize)]
+pub struct Mount {
+    /// A device name, but can also be a file or directory name for bind mounts or a dummy.
+    /// Path values for bind mounts are either absolute or relative to the bundle. A mount is a
+    /// bind mount if it has either bind or rbind in the options.
+    pub source: String,
+    /// Destination of mount point: path inside container. This value MUST be an absolute path.
+    pub destination: String,
+    /// The type of filesystem for the mountpoint.
+    pub fs_type: String,
+    /// Mount options for the mountpoint.
+    pub options: Vec<String>,
+    /// Optional device id for the block device when:
+    /// - the source is a block device or a mountpoint for a block device
+    /// - block device direct assignment is enabled
+    pub device_id: Option<String>,
+    /// Intermediate path to mount the source on host side and then passthrough to vm by shared fs.
+    pub host_shared_fs_path: Option<String>,
+    /// Whether to mount the mountpoint in readonly mode
+    pub read_only: bool,
+}
+
+impl Mount {
+    /// Get size of mount options.
+    pub fn option_size(&self) -> usize {
+        self.options.iter().map(|v| v.len() + 1).sum()
+    }
+}
+
+/// Check whether a mount type is a marker for Kata specific volume.
+pub fn is_kata_special_volume(ty: &str) -> bool {
+    ty.len() > KATA_VOLUME_TYPE_PREFIX.len() && ty.starts_with(KATA_VOLUME_TYPE_PREFIX)
```

Comment:
What if `ty == "kata: "` or `"kata:random-junk-here`? Same comment for functions below.

Reply:
The "kata:" prefix is added by plugins or the runtime itself, so we trust the content. And the is_kata_special_volume() is used to detect special marked cases, caller interested in the succeeding content should validate it before using the content.